definitions.h

```diff
@@ -47,6 +38,131 @@ struct Vertex
     double w;
 };
 
+/******************************************************
+ * CLass to take care of Matrices
+ *****************************************************/
+class Matrix
+{
+    public:
+    double matrix[4][4] = {{0,0,0,0},
```

Comment:
You should make this more dynamic than just a 4*4 matrix. Part of the assignment was being able to handle different sizes matrices